llvm/lib/Target/RISCV/MCTargetDesc/RISCVAsmBackend.cpp

```diff
@@ -309,9 +309,12 @@ bool RISCVAsmBackend::relaxAlign(MCFragment &F, unsigned &Size) {
   // Use default handling unless linker relaxation is enabled and the alignment
   // is larger than the nop size.
   const MCSubtargetInfo *STI = F.getSubtargetInfo();
-  if (!STI->hasFeature(RISCV::FeatureRelax))
+  if (!hasRelaxEver(F.getParent()) && !STI->hasFeature(RISCV::FeatureRelax))
```

Comment:
I assume `STI->hasFeature(RISCV::FeatureRelax)` is usually true so we should check it first before doing the set lookup